src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -172,4 +224,77 @@ core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createCDCTexture(const
     viewParams.subresourceRange.layerCount = 1u;
 
     return ICPUImageView::create(std::move(viewParams));
+}
+
+//! Explicit instantiations
+template core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createIESTexture(const std::execution::sequenced_policy&, const size_t&, const size_t&) const;
+// template core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createIESTexture(const std::execution::parallel_policy&, const size_t&, const size_t&) const;
+
+core::smart_refctd_ptr<asset::ICPUImageView> CIESProfile::createIESTexture(const size_t& width, const size_t& height) const
+{
+    return createIESTexture(std::execution::seq, width, height); 
+}
+
+template<class ExecutionPolicy>
+bool CIESProfile::flattenIESTexture(ExecutionPolicy&& policy, const IES_STORAGE_FORMAT& avgEmmision, core::smart_refctd_ptr<asset::ICPUImageView> inoutIES, const float flatten)
+{
+    if constexpr (policy._Parallelize)
+        static_assert(false, "meh it appears that filters with par policy are broken on this branch, use seq policy till fixed");
+
+    const bool inFlattenDomain = flatten >= 0.0 && flatten < 1.0; // [0, 1) range for blend equation, 1 is invalid
+
+    if (!inFlattenDomain)
+        return false;
+
+    auto outIES = inoutIES->getCreationParameters().image;
```

Comment:
don't do the `flatten` handling in the Loader!

Reply:
another reason not to do it here, is that you can log failures properly and skip certain checks (like format)